cl/search/models.py

```diff
@@ -2144,6 +2170,77 @@ class Meta:
         ordering = ["position"]
 
 
+class Courthouse(models.Model):
```

Comment:
For the various fields here, where possible, we should make them more similar to the fields in `AttorneyOrganization` (which also has address information). A few tweaks:

 - Use `USPostalCodeField` and `USZipCodeField`
 - Change the name of the fields to match the fields in `AttorneyOrganization` (address1 instead of address_line_1, etc.)
 - Use TextFields — They perform the same in postgresql and have unlimited length.

And semgrep is right. We shouldn't make text or char fields nullable. We do it sometimes, when we need uniqueness constraints, but very rarely. What makes you think it's our normal practice?

Reply:
Thanks.  I'll switch zip code over - but on second pass thru I think we should actually leave the state codes as is and add in the following options 

OBSOLETE_STATES = (
    ('CM', _('Commonwealth of the Northern Mariana Islands')),  # Is now 'MP'
    ('CZ', _('Panama Canal Zone')),  # Reverted to Panama 1979
    ('PI', _('Philippine Islands')),  # Philippine independence 1946
    # Became the independent COFA states + Northern Mariana Islands 1979-1994
    ('TT', _('Trust Territory of the Pacific Islands')),
)
that is not part of the `USPS_CHOICES` and we have 195 opinions in the Panama Canal Zone in the db.

Reply:
Fun. I think you should be able to split the baby with something like:

```
state = USPostalCodeField(choices=USPS_CHOICES + OBSOLETE_STATE_CHOICES)
```